test/benchmark/serialization_benchmark.rb

```diff
@@ -0,0 +1,88 @@
+require_relative 'benchmark_helper'
+module ActionController
+  module Serialization
+    class SerializerTest < ActionController::TestCase
+      class PostController < ActionController::Base
+        POST =
+          begin
+            comment = Comment.new(id: 1, body: 'ZOMG A COMMENT')
+            author  = Author.new(id: 1, name: 'Joao Moura.')
+            Post.new(id: 1, title: 'New Post', blog: nil, body: 'Body', comments: [comment], author: author)
+          end
+
+        def render_with_caching_serializer
+          render json: POST, adapter: :json
+        end
+
+        def render_with_non_caching_serializer
+          render json: POST, serializer: CachingPostSerializer, adapter: :json
+        end
+      end
+
+      tests PostController
+
+      def test_render_benchmark
+        cache_on!(true)
+        warmup
+        request_loop(:render_with_non_caching_serializer, n_times)
+      end
+
+      private
+
+      def request_loop(action, times)
+        ActionController::Base.cache_store.clear
+        i = 0
+        while i < times
+          get action
+          i += 1
+        end
+      end
```

Comment:
If I were benchmarking on one branch, I'd put the benchmark code in here.  But, because we're also comparing branches, I haven't settled on where I think the benchmarking should happen: in the test, or in the tool?